include/boost/geometry/core/coordinate_dimension.hpp

```diff
@@ -100,26 +97,26 @@ struct dimension
 \brief assert_dimension, enables compile-time checking if coordinate dimensions are as expected
 \ingroup utility
 */
-template <typename Geometry, int Dimensions>
+template <typename Geometry, std::size_t Dimensions>
 constexpr inline void assert_dimension()
 {
-    BOOST_STATIC_ASSERT(( static_cast<int>(dimension<Geometry>::value) == Dimensions ));
+    BOOST_STATIC_ASSERT(( dimension<Geometry>::value == Dimensions ));
```

Comment:
Is `GEOMETRY` missing here?

Reply:
This is the `static_assert` implemented outside Geometry. It only requires the condition while ours requires message and dependent types. Here the condition is used as the message. I left it as it was because I was only getting rid of `MPL_ASSERT`. `BOOST_STATIC_ASSERT` expands to `static_assert` in C++11.